lte/gateway/c/core/oai/tasks/sgw_s8/sgw_s8_state_manager.cpp

```diff
@@ -110,16 +117,27 @@ void SgwStateManager::free_state() {
     return;
   }
 
-  if (hashtable_ts_destroy(state_ue_ht) != HASH_TABLE_OK) {
-    OAI_FPRINTF_ERR(
-        "An error occurred while destroying SGW s11_bearer_context_information "
-        "hashtable");
+  if (s8_state_teid_map.map) {
+    if (s8_state_teid_map.destroy_map() != PROTO_MAP_OK) {
+      OAILOG_ERROR(LOG_SGW_S8,
+                   "An error occurred while destroying "
+                   "temporary_create_session_procedure_id_map ");
+    }
   }
+  s8_state_teid_map.map = nullptr;
```

Comment:
Is there a reason that you can not set `map` to `nullptr` in function `destroy_map()`?

Reply:
I had missed it. Thanks for bringing into notice